src/components/ImageUploader/src/ImageUploader.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+	<div :class="$s.ImageUploaderContainer">
+		<image-picker
+			v-if="canUploadImage"
+			:class="$s.ImageUploaderItem"
+			:multiple="canUploadMultiple"
+			:accept="accept"
+			@selectImages="selectImages"
+		/>
+		<image-selection
+			v-for="image of images"
+			:key="image.id"
+			:image="image"
+			:class="$s.ImageUploaderItem"
+			@removeImage="removeImage"
+		/>
+	</div>
+</template>
+
+<script>
+import ImagePicker from './ImagePicker.vue';
+import ImageSelection from './ImageSelection.vue';
+import { IMAGE_SELECTOR_STATUSES } from './constants';
+
+const MAX_PROGRESS = 100;
+const ID_INCREMENT = 1;
+const NO_MORE_IMAGES_COUNT = 0;
+
+export default {
+	name: 'MImageUploader',
+
+	components: {
+		ImagePicker,
+		ImageSelection,
+	},
+
+	props: {
+		/**
+		 * List of images that have been selected
+		 */
+		images: {
+			type: Array,
+			default: () => [],
+		},
+		// Not disabling and breaking the line causes the doc generator to break.
+		/* eslint-disable max-len */
+		/**
+		 * Function called to trigger an upload. Called immediately on image selection, provided max size and max number image constraints are met.
+		*/
+		/* eslint-enable max-len */
+		uploadHandlerFn: {
```

Comment:
Do you have a use case in mind for this? Could we get by with just emitting an event when the file input fires it's change event?

Reply:
Yeah the main thing I was trying to go for was to abstract away managing the progress and pending/complete/error states. If we use just an emitter, the consumer would be responsible for patching all those values and would add a bunch of overhead.
This standardizes that process while still providing enough flexibility to the consumer of this component to interact with their API.

Reply:
so one thing I've learned is that this function is completely optional, you just miss out on the realtime data (progress) from the server. 

Then just subjective opinion here, but `uploadHandler` is just as meaningful and more concise. But I'm wondering if we want to reference server / service in the name.